### PR TITLE
feat: single user (#1197)

### LINKED ISSUE #1197: [FEATURE] Allow creating LinkFree instances for single person
### Description

For now, if you want to host LinkFree, you have to host the whole profiles database.

If you want to host only an instance for yourself (like I did there this morning for testing purposes: https://links.piers.tech/pierstoval), you will have to remove the rest of the database.

I haven't digged into the code source yet, but would it be possible to add a switch somewhere to specify the **only** profile we would want to see?

It would need these changes:
* Disable `/{profile}/` route
* Make `/` route show the configured profile
* Remove the `⬅` link on the top left of the site
* Since this is for self-hosting purposes, the `Share` links must be updated too to use the current domain instead of `linkfree.eddiehub.org`

WDYT?

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by eddiejaoude**
## Fixes Issue

- Single user mode
- Update `src/config/user.json` with the username
- Still must follow the previous steps to add the user file
- Profile is displayed on root `/` and back nav is hidden

Closes #1197 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [ ] NEW tests are needed
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2022-03-31 at 09 44 20

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/1210"></a>

### CODE REVIEW

**Review by krshdev — APPROVED** [review]
Looks great to me 👍

**Review by schmelto — APPROVED** [review]
LGTM!

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good

**Comment by Pierstoval** [comment]
Tested and approved here 👍

**Comment by schmelto** [comment]
> Tested and approved here 👍

Hey @Pierstoval you can review PRs and approving in GitHub also in this way.
image

This shows also in the PR that you approve the PR see my approval earlier:
image


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good

File: src/App.js [hunk 1]
```diff
@@ -10,18 +10,23 @@ import Footer from './Components/Footer'
...
       <div className="m-2 md:m-4">
-        <Switch>
-          <Route path="/:username">
-            <User />
-          </Route>
-          <Route path="/">
-            <Home />
-          </Route>
-        </Switch>
+        {user.username && <User singleUser={user} />}
```
Comment (krshdev) [inline src/App.js:19]: Had a look at this from the desktop, I have a question, if someone visits `${site}/a_page_that_doesnot_exist`, will they not receive a 404 or redirection?

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

Comment (Pierstoval) [reply src/App.js:19]: Currently no, but they shouldn't do it anyway 🙈

**Review by Pierstoval — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (eddiejaoude) [reply src/App.js:19]: Good spot, as there is only 1 url, tbh I think catching all here and not showing a 404 is probably a good thing IMHO

**Comment by eddiejaoude** [comment]
Ok I will merge as it has a few reviews, thank you all ❤️

**Comment by eddiejaoude** [comment]
Oops! This really needs tests though, green squares anyone?

**Pull request merged**